src/v/cluster/types.h

```diff
@@ -409,7 +420,18 @@ struct begin_tx_reply : serde::envelope<begin_tx_reply, serde::version<0>> {
 
     friend std::ostream& operator<<(std::ostream& o, const begin_tx_reply& r);
 
-    auto serde_fields() { return std::tie(ntp, etag, ec); }
+    auto serde_fields() { return std::tie(ntp, etag, ec, topic_revision); }
+
+    void serde_read(iobuf_parser& in, const serde::header& h) {
```

Comment:
hmm, i kind of imagined that when a type outgrew `serde_fields` that it would replace it with explicit read/write methods, primarily because now it is ambiguous which one takes precedence. but that may not be optimal either. i wonder: would be cool if we could now mark `serde_fields` as `const`?

Reply:
yeah, i think `serde_fieds`should be enough to handle this in a transparent way

Reply:
yep, it should be handled by the serde_fields, I'll undo this part.